### ISSUE TITLE
Lifecycle Hooks

### ISSUE THREAD

**Issue by Nightcro** [timdeschryver/ngrx-immer#17]
Hello, 

I have a problem with the lifecycle hooks of ngrx  on component stores.
I am using ImmerComponentStore to create a store for a component and I want to use the  OnStateInit hooks.
The [guide](https://ngrx.io/guide/component-store/lifecycle) of NgRx says that I need to provide the store with ` provideComponentStore`, but whenever I do this for  a ImmerComponentStore , I get this error:

```
Error: immer.component.ts:14:37 - error TS2345: Argument of type 'typeof ImmerStore' is not assignable to parameter of type 'Type<ComponentStore<object>>'.
  Construct signature return types 'ImmerStore' and 'ComponentStore<object>' are incompatible.
    The types of 'updater' are incompatible between these types.
      Types of parameters 'updaterFn' and 'updaterFn' are incompatible.
        Type 'object' is not assignable to type 'void | MeetingState'.

14   providers: [provideComponentStore(ImmerStore)],

```

```
@Component({
  // ... other
  providers: [
    provideComponentStore(ImmerStore)
  ]
})
```

**Comment by timdeschryver** [comment]
I think we need to also add a `provideImmerComponentStore` method within this library.
Do you want to create a Pull Request for this?

**Comment by Nightcro** [comment]
Unfortunately, I do not have the experience to do that or the time yet. Fairly new to angular and typescript, so I would leave it to someone else with more experience and time.

For my problem, I managed to find a workaround for it.

**Comment by yevit** [comment]
I've just run into the same issue and opened a PR for it

**Comment by timdeschryver** [comment]
Closed by #18
